docs/mine/lotus/dynamic-retrieval-pricing.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Dynamic retrieval pricing
+description: Lotus allows you to set different policies to calculate the quoted price of a retrieval deal. The two available policies are Default, and External. 
+---
+
+# Dynamic retrieval pricing
+
+Lotus allows you to set different policies to calculate the quoted price of a retrieval deal. The two available policies are _Default_, and _External_:
+
+| Policy name | Description |
+| --- | --- |
+| Default | Lotus is shipped with an in-built default pricing policy that offers free retrievals for Verified Unsealed deals. Miners can still charge clients for data transfer if they wish to by turning off free data transfer for verified deals in the Miner config. Miners can also charge clients for Unsealing if they don't have an Unsealed sector file to retrieve the deal payload from. |
+| External | Miners can configure an external pricing script for Lotus to use instead of the default pricing policy to price retrieval deals. The script will be called with the relevant deal parameters, and the output quote will be used by Lotus to price deals. |
+
+## Default policy
+
+The default pricing policy uses the price configured in the `Ask Store`, set using the `lotus retrieval-deals set-ask` CLI command, to price all retrieval deals. However, it will not charge for data transfer if there exists a verified storage deal for the payload being retrieved. This behavior can be turned off by setting the `VerifiedDealsFreeTransfer` flag to `false` in the `DealMaking.RetrievalPricing` section of the config:
+
+```yaml
+[Dealmaking.RetrievalPricing]
+Strategy = "default"
+[Dealmaking.RetrievalPricing.Default]
+VerifiedDealsFreeTransfer = false
+```
+
+It will also not charge for Unsealing if we already have an Unsealed sector file containing the retrieval payload.
+
+## External policy
+
+Users can configure an external pricing script, similar to the deal filter mechanism, that takes a JSON marshaled `PricingInput` as the input and outputs a JSON marshaled `Ask`, also known as _the quote_. The `PricingInput` struct is defined as follows:
```

Comment:
@raulk - does _JSON marshaled_ just mean _JSON formatted_? Or is marshaling something different?